supervisor/machine_solaris.go

```diff
@@ -0,0 +1,11 @@
+package supervisor
+
+type Machine struct {
+	Cpus   int
+	Memory int64
+}
+
+func CollectMachineInformation() (Machine, error) {
```

Comment:
Yes there is. 
For now this can be done in one of two ways:
1. Once https://github.com/docker/docker/pull/22091 is approved, docker/docker/pkg/sysinfo should be able to provide the necessary information.
2. Its not very complex code, it could be implemented locally until gosigar support is resolved.
 
We will not be able to get this information via gosigar in the short term. gosigar does not have Solaris support and as we are not yet members of cloudfoundry we will not be able to add it until due process has been completed for that.